frontend/src/pages/Wallet.js

```diff
@@ -0,0 +1,149 @@
+
+import {
+	Button,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import appwriteApi from "../api/appwriteApi";
+import { useHistory } from "react-router-dom";
+import Grid from "@mui/material/Grid";
+import { useState } from "react";
+import detectEthereumProvider from "@metamask/detect-provider";
+import CenterFlexBoxMedium from "../components/CenterFlexBoxMedium";
+const { utils } = require( "ethers" );
+// import utils from "ethers";
+
+function AccountEntry({ data }) {
+	return 	<TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{data.pubAddress}
+		</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{utils.formatEther(data.balance)}
+		</TableCell>
+	</TableRow>;
+}
+
+function EthAccountsDetail({ publicAddresses }) {
+	const [detailData, setDetailData] = useState([]);
+
+	async function fetchAccountsDetails(publicAddresses) {
+		const provider = await detectEthereumProvider();
+		if (!provider) {
+			console.error("no ethereum provider");
+			return;
+		}
+		let accDetail = [];
+		const length = publicAddresses.length;
+		for (let i=0; i<length; i++) {
+			let pubAddress = publicAddresses[i];
+			const balance = await provider.request({ 
+				method: "eth_getBalance", 
+				params: [pubAddress, "latest"] 
+			});	// "Wei" balance in hex format, like 0x5a4e1804f198f1d99.
+			if (balance) {
+				accDetail.push({ 
+					pubAddress: pubAddress,
+					balance: balance
+				});
+			}
+		}
+		setDetailData(accDetail);
+	}
+
+	useEffect(() => {
+		fetchAccountsDetails(publicAddresses);
+	}, []);
+
+	return <TableContainer  style={{ color: "white" }}>
+		<Table>
+			<TableBody>
+				<TableRow>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>Account (Public Address)</TableCell>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>Balance (Eth)</TableCell>
+				</TableRow>
+				{detailData.map((value, key) => {
+					return <AccountEntry key={key} data={value}/>;
+				})}
+			</TableBody>
+		</Table>
+	</TableContainer>;
+}
+
+export default function Wallet({ user, setUser }) {
+	const [publicAddresses, setPublicAddresses] = useState("");
+	const history = useHistory();
+
+	const routeChange = (path) =>{
+		history.push(path);
+	};
+
+	useEffect(() => {
+		appwriteApi.getAccount()
+			.then(res => setUser(res));
+	}, []);
+
+	if (!user){
+		routeChange("/");
+		return <></>;
+	}
+
+	/* TODO: Fetch public address for this user from backend */
+
+	/* Handle button connect to MetaMask */
+	const handleAddMetaMask = async () => {
+		const ethereum = await detectEthereumProvider();
+		// Ref: https://docs.metamask.io/guide/rpc-api.html#table-of-contents
+		let pubAddr = 0;
+		if (ethereum) {
+			const accounts = await ethereum.request({ method: "eth_requestAccounts" });
+			const account = accounts[0];
+			pubAddr = account;
+			if (publicAddresses) {
```

Comment:
```suggestion
			setPublicAddresses( (publicAddresses ?? []) .push (publicAddress) )
```
Also possible `publicAddresses ??= []` .

Reply:
The "React way" of doing this would be something like this (not tested right now):

```javascript
setPublicAddresses(publicAddressesPrev => [...publicAddressesPrev, pubAddr])
```

And `const [publicAddresses, setPublicAddresses] = useState("");``should no be initialized as a String, if it is not a String, but as an empty list, i.e. []. In the code below (line 140), a length > 1 check instead of null check should be done.